xgboost_ray/matrix.py

```diff
@@ -227,6 +227,10 @@ def _split_dataframe(
         `label_upper_bound`
 
         """
+        # sort dataframe by qid if exists (required by DMatrix)
+        if self.qid and not local_data[self.qid].is_monotonic:
```

Comment:
```suggestion
        if self.qid is not None and not local_data[self.qid].is_monotonic:
```
should fix test failures!